source/common/ssl/ssl_socket.h

```diff
@@ -69,25 +69,32 @@ class SslSocket : public Network::TransportSocket,
 
 class ClientSslSocketFactory : public Network::TransportSocketFactory {
 public:
-  ClientSslSocketFactory(const ClientContextConfig& config, Ssl::ContextManager& manager,
+  ClientSslSocketFactory(ClientContextConfigPtr config, Ssl::ContextManager& manager,
                          Stats::Scope& stats_scope);
 
   Network::TransportSocketPtr createTransportSocket() const override;
   bool implementsSecureTransport() const override;
 
 private:
+  Ssl::ContextManager& manager_;
+  Stats::Scope& stats_scope_;
+  ClientContextConfigPtr config_;
```

Comment:
Why do we need to retain those? IT doesn't seem that those are used anywhere?

Reply:
It will be used in a later PR to create a new context when context_config is updated with new secret.

Reply:
Here is the code to use it
```
void ServerSslSocketFactory::onConfigUpdate() {
  ssl_ctx_ = manager_.createSslServerContext(stats_scope_, *config_, server_names_);
}

```

Reply:
OK, thanks!